File: API/Models/Game.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json;
using System.Text.Json.Serialization;

namespace API.Models
{
    [Table("games")]
    public class Game
    {
        [Key] 
        [Column("id")]
        public int Id { get; set; }
        [Column("team1")]
        public int Team1 { get; set; }
        [Column("team2")]
        public int Team2 { get; set; }
        
        [Column("goals1")]
        public int Goals1 { get; set; }
        [Column("goals2")]
        public int Goals2 { get; set; }

        [Column("status", TypeName = "text")]
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public GameStatus Status { get; set; }

        [Column("vars")]
        public JsonDocument? Vars { get; set; } 

        [Column("createdate")]
        public DateTime CreateDate { get; set; }

        [Column("completedate")]
        public DateTime? CompleteDate { get; set; }
        
        [Column("modifydate")]
        public DateTime ModifyDate { get; set; }
    }

    public enum GameStatus 
    {
        NotStarted,
        Playing,
        Completed,
        Cancelled
    }
}

File: client/src/models/Game.ts
import { convertToTeam, Team } from "./Team";
import { convertDataToUserList, User } from "./User";

export type Game = {
    Id: number;        
    Team1: Team;
    Team2: Team;
    Goals1: number;
    Goals2: number;    
    Vars: any;
    CreateDate: Date;
    ModifyDate: Date;
    CompleteDate: Date;
    Status: GameStatus
}

export enum GameStatus {
    NotStarted = "NotStarted",
    Playing = "Playing", 
    Completed = "Completed",
    Cancelled = "Cancelled"
}

export const convertToGame = (data : any) => {
    var game = {} as Game;    
    game.Id = parseInt(data.id);
    game.Team1 = data.team1 != null ? convertToTeam(data.team1) : {} as Team;
    game.Team2 = data.team2 != null ? convertToTeam(data.team2) : {} as Team;
    game.Goals1 = parseInt(data.goals1);
    game.Goals2 = parseInt(data.goals2);
    game.Vars = data.vars;    
    game.CreateDate = new Date(data.createDate);
    game.ModifyDate = new Date(data.modifyDate);
    game.CompleteDate = new Date(data.completeDate);
    game.Status = data.status as GameStatus;    
    return game;
}

export const convertDataToGameList = (listData : any) => {
    let list = [] as Game[];
    if (listData instanceof Array) {
        for (let i = 0; i < listData.length; i++) {				
            const o = convertToGame(listData[i]);
            if (o != null && o.Id != null) {
                list.push(o);
            }
        }
    }
    return list;
};

export const isValidGame = (game: Game) => {    
    if (game == null || game.Id == null || game.Team1 == null || game.Team2 == null) return false;    
    return true;
}